features/step_definitions/migration/migration_steps.rb

```diff
@@ -0,0 +1,128 @@
+    s.name == @instance_name and delete_old_instances(s.id)
+  end
+  # create a new instance
+  new_instance = control_node.exec!("openstack server create -f shell --flavor m1.smaller --image jeos --security-group default -c id #{@instance_name}")
+  @new_instance_id = new_instance.output[4, 36]
+  puts ("New instance id: #{@new_instance_id}")
+end
+
+When(/^I create an Xen instance$/) do
+  def delete_old_instances(server_id)
+    puts("old instance #{server_id } found - deleting")
+    control_node.exec!("openstack server delete #{server_id}")
+  end
+  ## TODO add --key-name zkubala
+  # clean old instances
+  @servers = control_node.openstack.server.list
+  @instance_name = "xen_mig_instance"
+  @servers.each do |s|
+    s.name == @instance_name and delete_old_instances(s.id)
+  end
+  # create a new instance
+  new_instance = control_node.exec!("openstack server create -f shell --flavor m1.smaller --image jeos --security-group default -c id #{@instance_name}")
+  @new_instance_id = new_instance.output[4, 36]
+  puts ("New instance id: #{@new_instance_id}")
+end
+
+When(/^Instance is running$/) do
+  ## TODO Raise an expection if in ERROR state 
+  # it may take some time before the instance turns active
+  wait_for "Checking that instance status is active", max: "120 seconds", sleep: "4 seconds" do
+    @instance_show = control_node.openstack.server.show(@new_instance_id)
+    break if @instance_show.status == "ACTIVE"
+  end
+end
+
+When(/^I migrate Xen instance$/) do
+  # get actual host of the instance and migrate it
+  @instance_host = @instance_show.send("os-ext-srv-attr:host")
+  puts "Instance running on: #{@instance_host}"
+  instance_migrate = control_node.exec!("openstack server migrate --shared-migration #{@new_instance_id}")
+
+  # it may take some time before the instance turns verify_resize
+  wait_for "Checking that instance status is verify_resize", max: "300 seconds", sleep: "10 seconds" do
+    instance_show = control_node.openstack.server.show(@new_instance_id)
+    if instance_show.status == "VERIFY_RESIZE"
+      puts "Migration finished, confirming..."
+      control_node.exec!("openstack server resize --confirm #{@new_instance_id}")
+    break
+    end
+  end
+end
+
+When(/^I migrate KVM instance$/) do
+  # get actual host of the instance and migrate it
+  @instance_host = @instance_show.send("os-ext-srv-attr:host")
+  puts "Instance running on: #{@instance_host}"
+  instance_migrate = control_node.exec!("openstack server migrate #{@new_instance_id}")
+
+  # it may take some time before the instance turns verify_resize
+  wait_for "Checking that instance status is verify_resize", max: "300 seconds", sleep: "10 seconds" do
+    instance_show = control_node.openstack.server.show(@new_instance_id)
+    if instance_show.status == "VERIFY_RESIZE"
+      puts "Migration finished, confirming..."
+      control_node.exec!("openstack server resize --confirm #{@new_instance_id}")
+    break
+    end
+  end
+end
```

Comment:
This step has exact same implementation as the one for Xen, it should work for both to parametrize the hypervisor name  in the step line in the feature file, e.g. ```When(/^I migrate "KVM" instance$/) do``` .